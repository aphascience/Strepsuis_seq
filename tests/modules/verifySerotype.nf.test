nextflow_process {

    name "Test Process VERIFY_SEROTYPE"
    script "modules/verifySerotype.nf"
    process "VERIFY_SEROTYPE"

    test("Should verify serotype against know SNPs") {

        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = ["test", "1", file("${projectDir}/tests/data/input_S999_R1_001.fastq.gz"), file("${projectDir}/tests/data/input_S999_R2_001.fastq.gz")]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
