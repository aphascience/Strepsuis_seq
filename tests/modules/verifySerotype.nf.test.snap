{
    "Should verify serotype against know SNPs": {
        "content": [
            {
                "0": [
                    [
                        "test",
                        "test_seroVerify.csv:md5,48f85ad9fd39b56feb35b1008e604ab9"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.7"
        },
        "timestamp": "2025-09-24T14:26:18.654919608"
    }
}